### PR TITLE
Added the forkjoin benchmark

### PR DESCRIPTION
**Pull request by kikofernandez**
Added the forkjoin benchmark

### CODE REVIEW

**Comment by kikofernandez** [comment]
I have assigned the reviewer as per [README.md in Savina](https://github.com/parapluu/encore/tree/development/src/tests/stress/savina)

**Comment by supercooldave** [comment]
I don't see any join in this framework, just forking.

**Comment by kikofernandez** [comment]
true, but that's the Savina name and the direct [translation of this benchmark](https://github.com/shamsimam/savina/blob/master/src/main/scala/edu/rice/habanero/benchmarks/fjcreate/ForkJoinAkkaActorBenchmark.scala)

**Comment by supercooldave** [comment]
This line
```
  AkkaActorState.awaitTermination(system)
```
seems to be `join`.

**Comment by kikofernandez** [comment]
reading the program carefully, I understand that they send messages and need to wait until all actors have finished their work. This is implicit in Encore, as the program cannot terminate unless all actors have performed their work

**Comment by supercooldave** [comment]
I won't fight this.
Were the original benchmarks done for Akka? If not, I wonder what they look like.

**Comment by kikofernandez** [comment]
I didn't get the comment at first and we have discussed this offline. 
The question refers to whether other libraries implement this in the same way. 
If we look at the [benchmark using Habanero](https://github.com/shamsimam/savina/blob/master/src/main/scala/edu/rice/habanero/benchmarks/fjcreate/ForkJoinHabaneroActorBenchmark.scala), they do exactly the same thing I did (no `join` operation involved)

File: src/tests/stress/savina/5.fjcreate/ForkJoinConfig.enc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    this.N = N
+    this.C = C
+  end
+
+  def performComputation(theta: real): unit
+    val sint = EMBED (real)
+                 sin(#{theta});
+               END
+    val res = sint * sint
+    if (res <= 0) then
```
Comment (supercooldave) [inline src/tests/stress/savina/5.fjcreate/ForkJoinConfig.enc:17]: sin(0) * sin(0) = 0
Why is this an unrealistic value?

File: src/tests/stress/savina/5.fjcreate/Main.enc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import ForkJoinActor
+
+fun getFromMaybe(x: Maybe[int]): int
+  match x with
+    case Just(i) => i
+    case Nothing => abort("Error: no value in maybe type")
+  end
+end
+
+fun bool_from_int(x: int): bool
```
Comment (supercooldave) [inline src/tests/stress/savina/5.fjcreate/Main.enc]: why not just `not x == 0`

**Review by supercooldave — CHANGES_REQUESTED** [review]
This will be ready to go after the minor comments have been addressed, and after changing benchmark name from `5.fjcreate` to `5.ForkJoinCreate` (or something appropriate).

File: src/tests/stress/savina/5.fjcreate/Main.enc (hunk 2 above)
Comment (kikofernandez) [reply src/tests/stress/savina/5.fjcreate/Main.enc]: > operator 'not' is not defined for values of type 'int'

File: src/tests/stress/savina/5.fjcreate/ForkJoinConfig.enc (hunk 1 above)
Comment (kikofernandez) [reply src/tests/stress/savina/5.fjcreate/ForkJoinConfig.enc:17]: I didn't make this up :)
[Original code here](https://github.com/shamsimam/savina/blob/master/src/main/java/edu/rice/habanero/benchmarks/fjcreate/ForkJoinConfig.java#L47)

```scala
protected static void performComputation(final double theta) {
    final double sint = Math.sin(theta);
    final double res = sint * sint;
    //defeat dead code elimination
    if (res <= 0) {
        throw new IllegalStateException("Benchmark exited with unrealistic res value " + res);
    }
}
```

File: src/tests/stress/savina/5.fjcreate/Main.enc (hunk 2 above)
Comment (supercooldave) [reply src/tests/stress/savina/5.fjcreate/Main.enc]: Must be `not (x == 0)`

File: src/tests/stress/savina/5.fjcreate/ForkJoinConfig.enc (hunk 1 above)
Comment (supercooldave) [reply src/tests/stress/savina/5.fjcreate/ForkJoinConfig.enc:17]: Amazing! Someone doesn't know his math.
Okay, just leave it.

**Comment by kikofernandez** [comment]
i have removed the function `bool_from_int, it was not used

**Pull request merged**